challenges/mergesort/mergesort.py

```diff
@@ -0,0 +1,33 @@
+def mergesort(lst):
+    """Take unsorted list and return it sorted."""
+    if type(lst) is not list:
+        raise TypeError('argument must be of type <list>')
+
+    left = lst[:len(lst)//2]
+    right = lst[len(lst)//2:]
+    if len(left) > 1:
+        left = mergesort(left)
+    if len(right) > 1:
+        right = mergesort(right)
+
+    slst = []
+    i, j = 0, 0
+    while i < len(left) and j < len(right):
+        if left[i] < right[j]:
+            slst.append(left[i])
+            i += 1
+        elif left[i] > right[j]:
+            slst.append(right[j])
+            j += 1
+        else:
+            slst.append(left[i])
+            slst.append(right[j])
+            i += 1
+            j += 1
+
+    for _ in range(i, len(left)):
+        slst.append(left[_])
+    for _ in range(j, len(right)):
+        slst.append(right[_])
```

Comment:
You can also use the `.extend()` method to do this with slices of the halves instead of iterating through them. Keep in mind that taking a slice will never raise an `IndexError` even if the index is past the end of the list.